### ISSUE TITLE
feature map size abnormal

### ISSUE THREAD

**Issue by veizei-K** [ultralytics/ultralytics#18429]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question
![image](https://github.com/user-attachments/assets/e8987192-4004-4e44-a8f8-f22b1b2bc914)
This is the size of the feature map I output from the first layer, but it's clearly not 320. The feature map size seems more like the size of the number of channels.
But when I tested it with the test script below, it worked fine.
```import torch
from ultralytics import YOLO

model = YOLO('yolo11s.yaml')

x = torch.randn(1, 3, 640, 640)  

outputs = {}

def hook(module, input, output):
    if isinstance(output, torch.Tensor): 
        outputs[module] = output.shape
    elif isinstance(output, list):  
        outputs[module] = [o.shape for o in output if isinstance(o, torch.Tensor)]
    else:  
        outputs[module] = str(type(output))

hooks = []
for name, module in model.model.named_modules():
    hooks.append(module.register_forward_hook(hook))

model(x)  

print("Feature Map Shapes:")
for module, shape in outputs.items():
    print(f"{module}: {shape}")

for h in hooks:
    h.remove()
```

### Additional
Including the feature map size of the SPPF, shouldn't it be 20x20? But it turns out to be 8x8.
![image](https://github.com/user-attachments/assets/00eda8cf-da4e-4db1-b666-1e158ffb856e)
"This is my training script."
```
from ultralytics import YOLO

# Load a model
model = YOLO("yolo11n.pt")

# Train the model
train_results = model.train(
    data="coco8.yaml",  # path to dataset YAML
    epochs=100,  # number of training epochs
    imgsz=640,  # training image size
    device=0,  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
)

```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @veizei-K, thank you for bringing this to our attention 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. It's possible that you may find some insights into layer behaviors and feature map sizes there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better investigate what might be going on. From the provided script and description, it seems you've already made good progress with debugging! For more clarity, please ensure you've shared all relevant custom parameters (if any), your `yolo11s.yaml` config details, and any potential modifications.

If this is a ❓ Question related to custom model behavior or design, we’d recommend sharing as much detailed information as possible. For example:
- The specific input size or data pipeline you're utilizing.
- Any alterations to the model architecture or configurations.

Additionally, please double-check you're using the latest version of `ultralytics`. Here's how you can update:

```bash
pip install -U ultralytics
```

Verifying your setup with the latest `ultralytics` ensures your issue hasn't already been addressed in a more recent release.

## Collaborate with the Ultralytics Community
Join us wherever you're comfortable for guidance and discussions:
- For real-time chat, join [Discord](https://discord.com/invite/ultralytics) 🎧.
- Prefer detailed forums? Visit [Discourse](https://community.ultralytics.com).
- Love Reddit? Check out our [Subreddit](https://reddit.com/r/Ultralytics).

## Verified Environments for YOLO
If you're running into persistent issues, try your experiment in one of our verified environments. Here are some options to test seamlessly:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker**: Find the [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) and access our optimized image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.  

## CI Verification
Our automated [Ultralytics CI system](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) <a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a> runs tests every 24 hours and on every commit to validate core functionality across platforms.

🔍 An Ultralytics engineer will review your report shortly. Meanwhile, feel free to dive into these resources and share further details if available. Thank you for helping us improve Ultralytics! 😊

**Comment by Y-T-G** [comment]
The first forward pass is due to initialization because it uses a dummy forward pass with smaller image size `(256,256)`:

https://github.com/ultralytics/ultralytics/blob/1051255c1e8de148fb85189da13059ae3937360e/ultralytics/nn/tasks.py#L336

**Comment by Y-T-G** [comment]
```python
In [10]: dummy = torch.randn(1,3,640,640)

In [11]: model.model.model[0](dummy).shape
Out[11]: torch.Size([1, 16, 320, 320])
```

**Comment by veizei-K** [comment]
I think I understand now. Before the model starts training, the image size doesn't seem to match the set size, but once it officially starts training, it returns to the correct size.
![image](https://github.com/user-attachments/assets/d4787a85-7d29-4154-a07f-03b8b4e44f7d)
This doesn't seem to be exactly the same as YOLOv8.

**Comment by veizei-K** [comment]
> The first forward pass is due to initialization because it uses a dummy forward pass with smaller image size `(256,256)`:
> 
> https://github.com/ultralytics/ultralytics/blob/1051255c1e8de148fb85189da13059ae3937360e/ultralytics/nn/tasks.py#L336

Yes, I think it’s better to directly use an image size of 640x640 for the first forward pass. I thought it was an issue with my settings or the model itself. YOLOv8 uses the default size of 640 for the first forward pass.

**Comment by glenn-jocher** [comment]
Thanks for pointing this out. The smaller image size during the first forward pass in YOLO11 is intentional and helps optimize initialization. While YOLOv8 defaults to 640x640, this difference in behavior is by design and doesn't impact training or inference. If you prefer consistency, you can modify the `model.model` initialization to use 640x640 directly.

**Comment by veizei-K** [comment]
> Thanks for pointing this out. The smaller image size during the first forward pass in YOLO11 is intentional and helps optimize initialization. While YOLOv8 defaults to 640x640, this difference in behavior is by design and doesn't impact training or inference. If you prefer consistency, you can modify the `model.model` initialization to use 640x640 directly.

For me, this makes debugging difficult because I don't know what the actual feature map size should be. Also, where can the option you mentioned, `model.model`, be set?

**Comment by glenn-jocher** [comment]
@veizei-K you can modify the model's first forward pass size by explicitly setting the input tensor dimensions before running it. This can be done by adjusting the dummy input size used during initialization in the source code. For debugging, you may prefer to set `model(x)` where `x` directly matches your training image size (e.g., 640x640). You can locate this initialization in the `ultralytics/nn/tasks.py` file. Feel free to adjust it to suit your needs, but note any changes will modify the default behavior.

**Comment by veizei-K** [comment]
> @veizei-K you can modify the model's first forward pass size by explicitly setting the input tensor dimensions before running it. This can be done by adjusting the dummy input size used during initialization in the source code. For debugging, you may prefer to set `model(x)` where `x` directly matches your training image size (e.g., 640x640). You can locate this initialization in the `ultralytics/nn/[tasks.py](http://tasks.py)` file. Feel free to adjust it to suit your needs, but note any changes will modify the default behavior.

@Y-T-G #https://github.com/ultralytics/ultralytics/issues/18429#issuecomment-2564275922
I found it, just need to modify here, thank you.
